scripts/cleanup/lib/analyze-js.sh

```
@@ -0,0 +1,174 @@
+analyze_frontend() {
+    echo "  → Analyzing frontend with knip..."
+
+    local output_file="$REPORT_DIR/knip-frontend-$TIMESTAMP.json"
+
+    if ! check_tool "npx" "npm install -g npm"; then
+        echo "    ⚠ npx not found, skipping frontend analysis"
+        FRONTEND_DEAD_CODE="{}"
+        return 1
+    fi
+
+    cd "$REPO_ROOT/frontend"
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)